File: Diagrammatize.Design.Pattern/code/cxx/FactoryMethod/Sample/idcard/IDCardFactory.cpp
#include "IDCardFactory.hpp"
#include "IDCard.hpp"

Product* IDCardFactory::createProduct(const std::string& owner) {
    return new IDCard(owner);
}

void IDCardFactory::registerProduct(Product* product) {
    owners.push_back(static_cast<IDCard *>(product)->getOwner());
}

const std::vector<std::string>& IDCardFactory::getOwners() {
    return owners;
}

File: Diagrammatize.Design.Pattern/code/python/FactoryMethod/Sample/idcard/IDCardFactory.py
from framework.Factory import Factory
from .IDCard import IDCard

class IDCardFactory(Factory):
    def __init__(self):
        self.owners = list()

    def createProduct(self, owner):
        return IDCard(owner)

    def registerProduct(self, product):
        self.owners.append(product.getOwner())

    def getOwners(self):
        return self.owners
